content/blog/webr-0-2-0/index.Rmd

```diff
@@ -0,0 +1,535 @@
+
+Other than interactive code blocks, like in the example earlier, the webR REPL app is perhaps the first thing that users new to webR will interact with. For this reason, we have spent some time working to improve the technical implementation and user experience of using the app. The app has been completely rewritten in the React web framework, replacing the older jQuery library. This allows for better component code organisation and more rapid development of features and updates.
+
+<a href="repl.png"><img alt="A screenshot the webR REPL app. The code to generate a ggplot, along with its output, is shown in the app." width="95%" src="repl.png"></a>
+
+### Code editor
+
+The app now comes with a tabbed code editor, allowing for easier editing and execution of R code. The editor integrates with the webR virtual filesystem (VFS), meaning that multiple R scripts can be opened, edited and saved and they will be available to the running Wasm R process.
+
+The editor pane is built upon the excellent [CodeMirror](https://codemirror.net) text editor, which provides most of the component's functionality. CodeMirror provides built-in support for syntax highlighting of R code, which is enabled by default when R source files are displayed.
+
+The editor is integrated with the currently running R process and automatic code suggestions are shown as you type, provided by R's [built in completion generator](https://stat.ethz.ch/R-manual/R-devel/library/utils/html/rcompgen.html). The suggestions are context sensitive and are aware of package and function names, valid arguments, and even objects that exist in the global environment.
+
+<a href="completion.png"><img alt="A screenshot of the editor component showing code completion results. One of the suggestions is a data set available in the global environment." width="70%" src="completion.png"></a>
+
+The running Wasm R process is also configured at initialisation to use the editor component as its display [pager mechanism](https://stat.ethz.ch/R-manual/R-devel/library/base/html/file.show.html). With this configuration in place running commands such as `?rnorm` in the app automatically opens a new read-only tab in the editor displaying R's built-in documentation.
+
+<a href="documentation.png"><img alt="A screenshot of the editor component showing built-in R documentation" width="80%" src="documentation.png"></a>
+
+### Plotting pane
+
+The plotting pane has been updated to take advantage of improvements in webR's HTML canvas graphic device, set as the default device as part of initialisation. In particular, multiple plots are now supported and older plots can be directly accessed using the previous and next buttons in the plotting toolbar. You can try this out with R's built in graphics demo, by running `demo(graphics)` and/or `demo(persp)`.
```

Comment:
Is it "graphic device" or "graphics device"? I'm not sure, but I feel like I see the latter more often, at least in other contexts.

Reply:
I was aiming for "graphics device", this was a typo. Thanks!